src/classic/element/ReactElementValidator.js

```diff
@@ -194,7 +194,10 @@ function validateChildKeys(node, parentType) {
         validateChildKeys(child, parentType);
       }
     }
-  } else if (ReactElement.isValidElement(node)) {
+  } else if (
+      typeof node === 'string' || typeof node === 'number' ||
+      ReactElement.isValidElement(node)
```

Comment:
Would it make sense to move these typechecks into `isValidElement`? I don't know what impact that has in the long run (vaguely sounds like something we've talked about before)